### PR TITLE
[mdatagen] support adding warnings to status table

### PR DESCRIPTION
**Pull request by codeboten**
This change allows components to specify warnings in metadata.yaml.

### CODE REVIEW

File: processor/transformprocessor/README.md [hunk 1]
```diff
@@ -169,10 +173,18 @@ Since the transform processor utilizes the OTTL's contexts for Traces, Metrics,
 In addition to OTTL functions, the processor defines its own functions to help with transformations specific to this processor:
 
 **Metrics only functions**
-- [convert_sum_to_gauge](#convert_sum_to_gauge)
-- [convert_gauge_to_sum](#convert_gauge_to_sum)
-- [convert_summary_count_val_to_sum](#convert_summary_count_val_to_sum)
-- [convert_summary_sum_val_to_sum](#convert_summary_sum_val_to_sum)
+- [Transform Processor](#transform-processor)
```
Comment (TylerHelmuth) [inline processor/transformprocessor/README.md:176]: I don't think this section should be modified.

Comment (codeboten) [reply processor/transformprocessor/README.md:176]: It definitely should not.... not sure what happened here

Comment (codeboten) [reply processor/transformprocessor/README.md]: Fixed

File: cmd/mdatagen/testdata/readme_with_warnings.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+| Stability                | [beta]   |
+| Supported pipeline types | metrics   |
+| Distributions            | [contrib] |
+| Warnings                 | [warning1](#warnings) |
+
+[beta]: https://github.com/open-telemetry/opentelemetry-collector#beta
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+<!-- end autogenerated section -->
+
+Some info about a component
```
Comment (atoulme) [inline cmd/mdatagen/testdata/readme_with_warnings.md:15]:
```suggestion
Some info about a component
### warnings
Some warning there.
```

**Comment by atoulme** [comment]
This is nice :)

**Review by atoulme — APPROVED** [review]

File: processor/transformprocessor/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: transform
+
+status:
+  type: processor
+  stability: alpha
+  pipelines: [logs, metrics, traces]
```
Comment (TylerHelmuth) [inline processor/transformprocessor/metadata.yaml:6]:
```suggestion
  pipelines: [traces, metrics, logs]
```

File: cmd/mdatagen/templates/readme.md.tmpl [hunk 4]
```diff
@@ -7,10 +7,13 @@
 | Supported pipeline types | {{ stringsJoin .Status.Pipelines "," }}   |
```
Comment (TylerHelmuth) [inline cmd/mdatagen/templates/readme.md.tmpl:7]: ", "
```suggestion
| Supported pipeline types | {{ stringsJoin .Status.Pipelines ", " }}   |
```

Comment (codeboten) [reply cmd/mdatagen/templates/readme.md.tmpl:7]: I'll commit this in a separate pr, as it changes many more readme's

File: cmd/mdatagen/testdata/readme_with_warnings.md (hunk 2 above)
Comment (codeboten) [reply cmd/mdatagen/testdata/readme_with_warnings.md:15]: Fixed

File: processor/transformprocessor/metadata.yaml (hunk 3 above)
Comment (codeboten) [reply processor/transformprocessor/metadata.yaml]: Fixed

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 4 above)
Comment (codeboten) [reply cmd/mdatagen/templates/readme.md.tmpl:7]: PTAL at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20244

**Pull request merged**